hopsworks-api/src/main/java/io/hops/hopsworks/api/iot/IotGatewayResource.java

```diff
@@ -0,0 +1,298 @@
+import io.hops.hopsworks.common.dao.project.team.ProjectRoleTypes;
+import io.hops.hopsworks.common.dao.project.team.ProjectTeam;
+import io.hops.hopsworks.common.dao.project.team.ProjectTeamFacade;
+import io.hops.hopsworks.common.dao.project.team.ProjectTeamPK;
+import io.hops.hopsworks.common.dao.user.UserFacade;
+import io.hops.hopsworks.common.dao.user.Users;
+import io.hops.hopsworks.common.kafka.KafkaController;
+import io.hops.hopsworks.common.project.ProjectController;
+import io.hops.hopsworks.exceptions.GatewayException;
+import io.hops.hopsworks.exceptions.KafkaException;
+import io.hops.hopsworks.exceptions.ProjectException;
+import io.hops.hopsworks.exceptions.UserException;
+import io.hops.hopsworks.jwt.annotation.JWTRequired;
+import io.hops.hopsworks.jwt.exception.DuplicateSigningKeyException;
+import io.hops.hopsworks.jwt.exception.SigningKeyNotFoundException;
+import io.swagger.annotations.ApiOperation;
+
+import javax.annotation.PostConstruct;
+import javax.ejb.EJB;
+import javax.ejb.TransactionAttribute;
+import javax.ejb.TransactionAttributeType;
+import javax.enterprise.context.RequestScoped;
+import javax.ws.rs.DELETE;
+import javax.ws.rs.GET;
+import javax.ws.rs.POST;
+import javax.ws.rs.PUT;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.SecurityContext;
+import javax.ws.rs.core.UriInfo;
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.security.NoSuchAlgorithmException;
+import java.util.Collections;
+import java.util.List;
+import java.util.logging.Logger;
+
+@RequestScoped
+@TransactionAttribute(TransactionAttributeType.NEVER)
+public class IotGatewayResource {
+  private static final Logger LOGGER = Logger.getLogger(IotGatewayResource.class.getName());
+  
+  @EJB
+  private IotGatewayBuilder iotGatewayBuilder;
+  @EJB
+  private ProjectController projectController;
+  @EJB
+  private ProjectFacade projectFacade;
+  @EJB
+  private IotGatewayFacade iotGatewayFacade;
+  @EJB
+  private UserFacade userFacade;
+  @EJB
+  private IotGatewayController iotGatewayController;
+  @EJB
+  private KafkaController kafkaController;
+  @EJB
+  private JWTHelper jWTHelper;
+  @EJB
+  private ProjectTeamFacade projectTeamFacade;
+  
+  private Project project;
+  
+  public IotGatewayResource setProject(Integer projectId) {
+    this.project = projectFacade.find(projectId);
+    return this;
+  }
+  
+  @PostConstruct
+  public void init() {
+  
+  }
+  
+  @ApiOperation(value = "Activate IoT feature")
+  @POST
+  @Path("activateIot")
+  @AllowedProjectRoles({AllowedProjectRoles.DATA_OWNER, AllowedProjectRoles.DATA_SCIENTIST})
+  @JWTRequired(acceptedTokens = {Audience.API}, allowedUserRoles = {"HOPS_ADMIN", "HOPS_USER"})
+  public Response activateIot(@Context SecurityContext sc) throws KafkaException, UserException, ProjectException {
```

Comment:
@kai-chi this belongs to the ProjectResource and the ProjectController.addService. It's better to have IoT as a new Project service and reuse the same code patterns we have for enabling services